Mathlib/Combinatorics/SimpleGraph/Clique.lean

```diff
@@ -655,4 +651,211 @@ theorem cliqueFinset_map_of_equiv (e : α ≃ β) (n : ℕ) :
+  by_contra nind
+  simp [SimpleGraph.IsIndepSet, Set.Pairwise] at nind
+  obtain ⟨j, avj, k, avk, _, ajk⟩ := nind
+  exact h {v, j, k} (is3Clique_triple_iff.mpr (by simp [avj, avk, ajk]))
+
+end IndepSet
+
+/-! ### N-Independent sets -/
+
+
+section NIndepSet
+
+variable {n : ℕ} {s : Finset α}
+
+/-- An `n`-independent set in a graph is a set of `n` vertices which are pairwise nonadjacent. -/
+@[mk_iff]
+structure IsNIndepSet (n : ℕ) (s : Finset α) : Prop where
+  isIndepSet : G.IsIndepSet s
+  card_eq : s.card = n
+
+/-- An `n`-independent set is an `n`-clique in the complement graph and vice versa. -/
+@[simp] theorem isNClique_compl : Gᶜ.IsNClique n s ↔ G.IsNIndepSet n s := by
+  rw [isNIndepSet_iff]
+  simp [isNClique_iff]
+
+/-- An `n`-independent set in the complement graph is an `n`-clique and vice versa. -/
+@[simp] theorem isNIndepSet_compl : Gᶜ.IsNIndepSet n s ↔ G.IsNClique n s := by
+  rw [isNClique_iff]
+  simp [isNIndepSet_iff]
+
+instance [DecidableEq α] [DecidableRel G.Adj] {n : ℕ} {s : Finset α} :
+    Decidable (G.IsNIndepSet n s) :=
+  decidable_of_iff' _ (G.isNIndepSet_iff n s)
+
+end NIndepSet
+
+/-! ### Graphs without independent sets -/
+
+
+section IndepSetFree
+
+variable {n : ℕ}
+
+/-- `G.IndepSetFree n` means that `G` has no `n`-independent sets. -/
+def IndepSetFree (n : ℕ) : Prop :=
+  ∀ t, ¬G.IsNIndepSet n t
+
+/-- An graph is `n`-independent set free iff its complement is `n`-clique free. -/
+@[simp] theorem cliqueFree_compl : Gᶜ.CliqueFree n ↔ G.IndepSetFree n := by
+  simp [IndepSetFree, CliqueFree]
+
+/-- An graph's complement is `n`-independent set free iff it is `n`-clique free. -/
+@[simp] theorem indepSetFree_compl : Gᶜ.IndepSetFree n ↔ G.CliqueFree n := by
+  simp [IndepSetFree, CliqueFree]
+
+/-- `G.IndepSetFreeOn s n` means that `G` has no `n`-independent sets contained in `s`. -/
+def IndepSetFreeOn (G : SimpleGraph α) (s : Set α) (n : ℕ) : Prop :=
+  ∀ ⦃t⦄, ↑t ⊆ s → ¬G.IsNIndepSet n t
+
+end IndepSetFree
+
+/-! ### Set of independent sets-/
+
+
+section IndepSetSet
+
+variable {n : ℕ} {s : Finset α}
+
+/-- The `n`-independent sets in a graph as a set. -/
+def indepSetSet (n : ℕ) : Set (Finset α) :=
+  { s | G.IsNIndepSet n s }
+
+variable {G}
+
+@[simp]
+theorem mem_indepSetSet_iff : s ∈ G.indepSetSet n ↔ G.IsNIndepSet n s :=
+  Iff.rfl
+
+end IndepSetSet
+
+/-! ### Independence Number-/
+
+
+section IndepNumber
+
+variable {α : Type*} {G : SimpleGraph α}
+
+/-- The maximal number of vertices of an independent set in a graph `G`. -/
+noncomputable def indepNum (G : SimpleGraph α) : ℕ := sSup {n | ∃ s, G.IsNIndepSet n s}
+
+@[simp] lemma cliqueNum_compl : Gᶜ.cliqueNum = G.indepNum := by
+  simp [indepNum, cliqueNum]
+
+@[simp] lemma indepNum_compl : Gᶜ.indepNum = G.cliqueNum := by
+  simp [indepNum, cliqueNum]
+
+theorem IsIndepSet.card_le_indepNum
+    [Fintype α] {t : Finset α} {tc : G.IsIndepSet t} : #t ≤ G.indepNum := by
+  simp [indepNum, ← isClique_compl, ← isNClique_compl] at *
```

Comment:
similarly you should adjust the `simp` calls here and below (the calls to `simp` in `cliqueNum_compl` and `indepNum_compl` close the goal, so they're fine!)